gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,634 @@
+          not hard-filtered.
+        - Any sample in `ht` that is hard-filtered will have a missing value for these
+          annotations.
+
+    These related filter annotations are also provided for release filtered samples
+    added to the input `ht` under a `release_relatedness_filters` struct:
+
+        - related: Whether the release filtered sample was filtered for
+          second-degree (or closer) relatedness in the final release.
+        - duplicate_or_twin: Whether the release filtered sample has a
+          duplicate or twin among all samples that are not hard-filtered or
+          outlier-filtered.
+        - parent_child: Whether the release filtered sample has a parent or
+          child among all samples that are not hard-filtered or outlier-filtered.
+        - sibling: Whether the release filtered sample has a sibling among all
+          samples that are not hard-filtered or outlier-filtered.
+        - Any sample in `ht` that is hard-filtered or outlier-filtered will have a
+          missing value for these annotations.
+
+    :param ht: Sample QC filter Table to add relatedness filter annotations to.
+    :param relationship_ht: Table with relationships annotations.
+    :param hard_filtered_expr: Boolean Expression indicating whether the sample was
+        hard filtered.
+    :param outlier_filtered_expr: Boolean Expression indicating whether the sample was
+        outlier filtered.
+    :return: Table with related filters added and Table with relationship and gnomad v3
+        overlap information.
+    """
+    rel_dict = {
+        "related": SECOND_DEGREE_RELATIVES,
+        "duplicate_or_twin": DUPLICATE_OR_TWINS,
+        "parent_child": PARENT_CHILD,
+        "sibling": SIBLINGS,
+    }
+    relationships = relationship_ht[ht.key]
+    relatedness_filters_ht = ht.annotate(
+        relatedness_filters=hl.struct(
+            **{
+                rel: get_relationship_filter_expr(
+                    hard_filtered_expr,
+                    hl.is_defined(related_samples_to_drop(release=False).ht()[ht.key]),
+                    relationships.relationships,
+                    rel_val,
+                )
+                for rel, rel_val in rel_dict.items()
+            }
+        ),
+        release_relatedness_filters=hl.struct(
+            **{
+                rel: get_relationship_filter_expr(
+                    hard_filtered_expr | outlier_filtered_expr,
+                    hl.is_defined(related_samples_to_drop(release=True).ht()[ht.key]),
+                    relationships.relationships_high_quality,
+                    rel_val,
+                )
+                for rel, rel_val in rel_dict.items()
+            }
+        ),
+    )
+
+    return relatedness_filters_ht
+
+
+def main(args):
+    """Merge the output of all sample QC modules into a single Table."""
+    hl.init(
+        spark_conf={"spark.memory.offHeap.enabled": "false"},
```

Comment:
Is this the same as initializing the cluster with no off heap mem? If so, nice!

Reply:
yeah, but forgot i can remove for this script